pkg/util/hostname.go

```diff
@@ -157,7 +157,7 @@ func GetHostnameData() (HostnameData, error) {
 	log.Debug("Trying to determine a reliable host name automatically...")
 
 	// if fargate we strip the hostname
-	if ecs.IsFargateInstance() {
+	if ecs.IsFargateInstance() || config.Datadog.GetBool("eks_fargate") {
```

Comment:
Should this be handled inside the `IsFargateInstance` ? Isn't eks+fargate still fargate ?

Reply:
This was my first idea :). 

The package "ecs" is only for the "ecs" environment, but "eks" is a completely different environment (kubernetes). So if we put this logic inside the "ecs" package we are missing things...
Also, I feel is was over eng to create a dedicated "eks" package just for this configuration check.
WDYT?